lib/utils.pm

```diff
@@ -1078,7 +1093,26 @@ sub set_hostname {
     assert_script_run "uname -n|grep $hostname";
     systemctl 'status network.service';
     save_screenshot;
-    assert_script_run "if systemctl -q is-active network.service; then systemctl reload-or-restart network.service; fi";
+
+    if (systemctl('is-active NetworkManager', ignore_failure => 1) == 0) {
+        my $conn_id = script_output 'basename $(ls -1 /etc/NetworkManager/system-connections | head -n 1) .nmconnection';
```

Comment:
maybe rather use something like `nmcli device show | grep GENERAL.DEVICE | grep -v "\Wlo$"` ?

Reply:
I had already written something slightly different in #17221. I guess the main thing is they both use `nmcli` to get the device list